core/src/main/scala/org/bitcoins/core/protocol/AddressFactory.scala

```diff
@@ -2,26 +2,31 @@ package org.bitcoins.core.protocol
 
 import org.bitcoins.core.config.NetworkParameters
 import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.crypto.StringFactory
 
-import scala.util.{Failure, Success, Try}
+import scala.util.Try
 
-abstract class AddressFactory[T] {
-
-  /** Attempts to create an address from the given String */
-  def fromString(str: String): Try[T]
+abstract class AddressFactory[T <: Address] extends StringFactory[T] {
 
   /** Same as fromString, but throws the exception */
-  def fromStringExn(str: String): T = fromString(str) match {
-    case Success(addr) => addr
-    case Failure(exn)  => throw exn
+  def fromStringExn(str: String): T = fromString(str)
+
+
+  @deprecated(s"Use fromScriptPubKeyT", since="2020-06-09")
+  def fromScriptPubKey(spk: ScriptPubKey, np: NetworkParameters): Try[T] = {
```

Comment:
shouldn't this just be changed to match the normal pattern that we have now where this returns a `T` and we can just rework all calls to it?